constantine/math/elliptic/ec_endomorphism_accel.nim

```diff
@@ -84,15 +67,11 @@ func decomposeEndo*[M, scalBits, L: static int](
       alphas[i].setZero()
     else:
       alphas[i].prod_high_words(babai(F)[i][0], scalar, w)
-    when babai(F)[i][1]:
-      # prod_high_words works like logical right shift
-      # When negative, we should add 1 to properly round toward -infinity
-      alphas[i] += One
```

Comment:
This is the main change. Papers that introduce the Babai's rounding unfortunately use shifts but do not go over the negative special case.